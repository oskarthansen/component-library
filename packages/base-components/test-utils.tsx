export * from '../../utils/base-test-utils';
