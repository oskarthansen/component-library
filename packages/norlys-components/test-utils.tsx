import 'jest-styled-components';
import React, { ComponentType, ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';
import { render as rtlRender, RenderOptions, RenderResult } from '../../utils/base-test-utils';
import lightTheme from './src/themes/light-theme';

const themedRender = (ui: ReactElement, options?: RenderOptions): RenderResult => {
  const Wrapper = ({ children }: { children: ReactElement }): ReactElement => {
    return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
  };

  return rtlRender(ui, { wrapper: Wrapper as ComponentType, ...options });
};

export * from '../../utils/base-test-utils';
export { themedRender, lightTheme };
