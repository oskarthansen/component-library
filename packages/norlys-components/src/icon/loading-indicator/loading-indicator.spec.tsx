import React from 'react';
import { themedRender } from '../../../test-utils';

import LoadingIndicator from './loading-indicator';

describe('<LoadingIndicator />', () => {
  describe('snapshot', () => {
    it('should match the snapshot', () => {
      const { container } = themedRender(<LoadingIndicator />);

      expect(container).toMatchSnapshot();
    });
  });
});
