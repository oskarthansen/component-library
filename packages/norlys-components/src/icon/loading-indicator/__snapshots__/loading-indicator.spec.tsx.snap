// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> snapshot should match the snapshot 1`] = `
.c0 {
  width: 20px;
  height: 20px;
  fill: currentColor;
  font-size: 24px;
  color: #FFFFFF;
}

.c1 {
  -webkit-animation: eoUyJr 0.8s linear infinite;
  animation: eoUyJr 0.8s linear infinite;
}

.c1 ellipse {
  fill: white;
}

<div>
  <svg
    class="c0 c1"
    color="white"
    viewBox="0 0 20 20"
  >
    <ellipse
      cx="18.0002"
      cy="9.99587"
      opacity="0.7"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="2.00024"
      cy="9.99587"
      opacity="0.3"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="16.0002"
      cy="15.9959"
      opacity="0.05"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="4.00415"
      cy="15.9959"
      opacity="0.2"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="16.0002"
      cy="3.99977"
      opacity="0.6"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="4.00415"
      cy="3.99977"
      opacity="0.4"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="10.0042"
      cy="17.9998"
      opacity="0.1"
      rx="1.99977"
      ry="1.99976"
    />
    <ellipse
      cx="10.0042"
      cy="1.99977"
      opacity="0.5"
      rx="1.99977"
      ry="1.99976"
    />
  </svg>
</div>
`;
